File: INIGO   SOLOMON/desk/AdditionClient.java
import org.apache.thrift.TException;
import org.apache.thrift.protocol.TBinaryProtocol;
import org.apache.thrift.protocol.TProtocol;
import org.apache.thrift.transport.TSocket;
import org.apache.thrift.transport.TTransport;

public class AdditionClient {
  public static void main(String [] args) {

   
    try {
      TTransport transport;
     
      transport = new TSocket("localhost", 9090);
      transport.open();

      TProtocol protocol = new  TBinaryProtocol(transport);
      AdditionService.Client client = new AdditionService.Client(protocol);

      perform(client);

      transport.close();
    } catch (TException x) {
      x.printStackTrace();
    } 
  }

  private static void perform(AdditionService.Client client) throws TException
  {
   
    int product = client.sum(3,5);
    System.out.println("3+5=" + sum);
  }
}

File: INIGO   SOLOMON/desk/Thrift try/AdditionClient.py
import sys
sys.path.append('../gen-py')

from tutorial import AdditionService
from tutorial.ttypes import *

from thrift import Thrift
from thrift.transport import TSocket
from thrift.transport import TTransport
from thrift.protocol import TBinaryProtocol

try:

	transport=TSocket.TSocket('localhost',9090)
	transport=TTransport.TBufferedTransport(transport)
	protocol=TBinaryProtocol.TBinaryProtocol(transport)
	client=AdditionService.Client(protocol)
	transport.open()
	sum=client.add(4,5)
	print '4+5=%d' %(sum)
	transport.close()
except Thrift.TException,tx:
	print '%s' %(tx.message)
